Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabLoader.cpp

```diff
@@ -277,6 +280,55 @@ namespace AzToolsFramework
             return !nestedTemplateReference->get().IsLoadedWithErrors();
         }
 
+        bool PrefabLoader::SanitizeLoadedTemplate(PrefabDomReference loadedTemplateDom)
+        {
+            if (!loadedTemplateDom)
+            {
+                return false;
+            }
+
+            Instance loadedPrefabInstance;
+            if (!PrefabDomUtils::LoadInstanceFromPrefabDom(loadedPrefabInstance, loadedTemplateDom->get()))
+            {
+                return false;
+            }
+
+            PrefabDom storedPrefabDom(&loadedTemplateDom->get().GetAllocator());
+            storedPrefabDom.SetObject();
+            if (!PrefabDomUtils::StoreInstanceInPrefabDom(loadedPrefabInstance, storedPrefabDom))
+            {
+                return false;
+            }
+
+            loadedTemplateDom->get().CopyFrom(storedPrefabDom, loadedTemplateDom->get().GetAllocator());
+            return true;
+        }
+
+        bool PrefabLoader::SanitizeSavingTemplate(PrefabDomReference savingTemplateDom)
```

Comment:
Both the functions do the same thing except pass a flag. Can we create a common 'sanitizeTemplate' function that accepts an optional flag?

Reply:
They're similar now but going forward we may want to add more scrubbing/sanitation code that is more specific to one over the other. Such as adding/removing components only required in the Editor. Or calling events on entities during save.

Reply:
Correct, will keep them up for now but I expect them to be refactored soon-ish.